### PR TITLE
4.x : Add features to test Mysql datasource

### PR DESCRIPTION
**Pull request by deepakrkris**
- [ ] - add mysql datasource for CI environment
- [ ] - migrate model to datasource
- [ ] - inject mock datasource only for NON-CI environment

### CODE REVIEW

**Comment by deepakrkris** [comment]
@slnode test please

File: test/helpers/setup-database.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      connection = mysql.createConnection({
+        user: 'root',
+        password: password,
+      });
+
+      console.log('Connecting');
+      connection.connect(next);
+    },
+    function createDatabase(next) {
+      console.log('Creating database %s', config.DATABASE);
```
Comment (superkhau) [inline test/helpers/setup-database.js]: all console.log to [debug](https://github.com/strongloop/loopback-datasource-juggler/blob/master/package.json#L44)

File: test/helpers/setup-database.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        '"@"localhost" IDENTIFIED BY "' +
+        config.PASSWORD + '"' +
+        ' WITH GRANT OPTION',
+        next);
+    },
+  ], function(err) {
+    connection.end();
+    if (err) {
+      console.error('Setup failed. %s', err);
+      process.exit(1);
```
Comment (superkhau) [inline test/helpers/setup-database.js]: you can just `throw new Error('Setup failed...)` to stop execution if err and console log in the other case

File: test/helpers/test-support.js [hunk 3]
```diff
@@ -65,3 +67,26 @@ function localInstall(cwd, cb) {
...
+    function(err, ds) {
+      if (err) return cb(err);
+      ds.connector = 'mysql';
+      ds.facetName = 'server';
+      ds.database = config.DATABASE;
+      ds.user = config.USER;
+      ds.password = config.PASSWORD;
+      ds.save(testsuite.getContext(templateName), cb);
+    });
+}
```
Comment (superkhau) [inline test/helpers/test-support.js]: indent is off

**Comment by deepakrkris** [comment]
@slnode test please

**Comment by deepakrkris** [comment]
@slnode test please

**Comment by deepakrkris** [comment]
@slnode test please

**Comment by rmg** [comment]
@slnode test please

**Comment by rmg** [comment]
The failed `[cis-jenkins] x64 && linux && nvm,4` and `[cis-jenkins] x64 && linux && nvm,6` builds were triggered before the changes from #446 were picked up by the repo scanner. They should be ignored.

I kicked off a rebuild which used the update job config which will include the required `[cis-jenkins] x64 && linux && nvm && dbs,6` and `[cis-jenkins] x64 && linux && nvm && dbs,4` statuses.

**Pull request merged**